### PR TITLE
Initial Azure.ResourceManager.ConfidentialLedger

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.ConfidentialLedger.Models
+{
+    /// <summary> AAD based security principal with associated Ledger RoleName. </summary>
+    public partial class AADBasedSecurityPrincipal
```
Comment (Yao725) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs:11]: Should be `AadBasedSecurityPrincipal`?

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            PrincipalId = principalId;
+            TenantId = tenantId;
+            LedgerRoleName = ledgerRoleName;
+        }
+
+        /// <summary> UUID/GUID based Principal Id of the Security Principal. </summary>
+        public string PrincipalId { get; set; }
+        /// <summary> UUID/GUID based Tenant Id of the Security Principal. </summary>
+        public string TenantId { get; set; }
```
Comment (Yao725) [inline sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs:32]: They should be `Guid`.

**Review by Yao725 — APPROVED** [review]
LGTM, left a few minor comments.

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs]: fixed

File: sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/confidentialledger/Azure.ResourceManager.ConfidentialLedger/src/Generated/Models/AADBasedSecurityPrincipal.cs]: fixed

**Pull request merged**